Sources/RealModule/ApproximateEquality.swift

```diff
@@ -0,0 +1,154 @@
+  /// Compares `self` and `other` for approximate equality with a specified
+  /// relative tolerance and implicit absolute tolerance.
+  ///
+  /// `true` if `self` and `other` are equal, or if they are finite and
+  /// ```
+  /// (self - other).magnitude ≤ relativeTolerance * scale
+  /// ```
+  /// where `scale` is
+  /// ```
+  /// max(self.magnitude, other.magnitude, .leastNormalMagnitude)
+  /// ```
+  ///
+  /// The default value of `relativeTolerance` is `.ulpOfOne.squareRoot()`,
+  /// which corresponds to expecting "about half the digits" in the computed results to be good.
+  /// This is the usual guidance in numerical analysis, if you don't know anything about the
+  /// computation being performed, but is not suitable for all use cases.
+  ///
+  /// Mathematical Properties:
+  /// -
+  /// - `isApproximatelyEqual(to:relativeTolerance:)` is _reflexive_ for
+  ///   non-exceptional values (such as NaN).
+  ///
+  /// - `isApproximatelyEqual(to:relativeTolerance:)` is _symmetric_.
+  ///
+  /// - `isApproximatelyEqual(to:relativeTolerance:)` is __not__ _transitive_.
+  ///   Because of this, approximately equality is __not an equivalence relation__,
+  ///   even when restricted to non-exceptional values.
+  ///
+  /// - For any point `a`, the set of values that compare approximately equal to `a` is _convex_.
+  ///   (Under the assumption that the `.magnitude` property implements a valid norm.)
+  ///
+  /// - `isApproximatelyEqual(to:relativeTolerance:)` is _scale invariant_,
+  ///   so long as no underflow or overflow has occured, and no exceptional value is produced
+  ///   by the scaling.
+  ///
+  /// - Parameters:
+  ///
+  ///   - other: The value to which `self` is compared.
+  ///
+  ///   - relativeTolerance: The tolerance to use for the comparison.
+  ///     Defaults to `.ulpOfOne.squareRoot()`.
+  ///
+  ///     This value should be non-negative and less than or equal to 1.
+  ///     This constraint on is only checked in debug builds, because a mathematically
+  ///     well-defined result exists for any tolerance, even one out of range.
+  @inlinable @inline(__always)
+  public func isApproximatelyEqual(
+    to other: Self,
+    relativeTolerance: Magnitude = Magnitude.ulpOfOne.squareRoot()
+  ) -> Bool {
+    return isApproximatelyEqual(
+      to: other,
+      absoluteTolerance: relativeTolerance * Magnitude.leastNormalMagnitude,
+      relativeTolerance: relativeTolerance
+    )
+  }
+  
+  /// Compares `self` and `other` for approximate equality with a specified tolerances.
+  ///
+  /// `true` if `self` and `other` are equal, or if they are finite and either
+  /// ```
+  /// (self - other).magnitude ≤ absoluteTolerance
+  /// ```
+  /// or
+  /// ```
+  /// (self - other).magnitude ≤ relativeTolerance * scale
+  /// ```
+  /// where `scale` is `max(self.magnitude, other.magnitude)`.
+  ///
+  /// Mathematical Properties:
+  /// -
+  /// - `isApproximatelyEqual(to:absoluteTolerance:relativeTolerance:)`
```

Comment:
Should these include a `norm` argument label? Or was it deliberately omitted because you don't introduce `norm` until later in the comment?

Reply:
Omitted because it simply becomes too verbose and fewer users should need to use it.

Reply:
(To be precise about what I mean by "fewer users should need to use it": the `norm` parameter doesn't matter for Real values, which is what I expect to be the overwhelming majority of uses. It only becomes relevant with Complex types or vectors / quaternions / matrices / etc.)